File: tlb/06/solutions/merge_whales.py
#!/usr/bin/python

import re, sys

current_count = 0
current_species = ""
for line in sys.stdin:
    m = re.search(r'(\d+)\s*(.+)$', line)
    if m:
        count = int(m.group(1))
        species = m.group(2)
        if species  == current_species:
            current_count += count
        else:
            if current_count:
                print(current_count + " " +current_species)
            current_count = count
            current_species = species
    else:
        print("Sorry couldn't parse: %s" % line)

if current_count:
    print(current_count + " " +current_species)

File: tlb/06/solutions/merge_whales.pl
#!/usr/bin/perl -w

$current_count = 0;
$current_species = "";

while ($line = <STDIN>) {
	if ($line =~ /(\d+)\s*(.+)\s*$/) {
		$count = $1;
		$species = $2;
		if ($species eq $current_species) {
			$current_count += $count;
		} else {
			print "$current_count $current_species\n" if $current_count;
			$current_count = $count;
			$current_species = $species;
		}
	} else {
		print "Sorry couldn't parse: $line\n";
	}
}

print "$current_count $current_species\n" if $current_count;
